internal/server/handlers.go

```diff
@@ -151,13 +146,16 @@ func (h *handler) loop() {
 		}
 
 		entries := h.swapEntries()
-		if err := h.send(entries); err != nil {
-			return
-		}
+
+		err := h.send(entries)
```

Comment:
Is it worth adding a comment mentioning that we `clear` the entries and return it to pool before returning the error so we don't skip the clearing? 

It doesn't seem too subtle what we're doing so I'm also fine leaving it implied

Reply:
Well I disagree with the statement that it's implied. No comment will ever be as explicit as actual code, since that's ultimately what gets executed! But yeah I think this code is clear enough not to require a comment